db/commitlog/commitlog.cc

```diff
@@ -1050,6 +1076,33 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
             write(out, uint64_t(0));
         }
 
+        buf.remove_suffix(buf.size_bytes() - size);
+
+        {
```

Comment:
in otherplaces we changed the convention from using scoped, bracketed code, to using `std::invoke()`
and coding the block as a function.

Reply:
Why? I see no benefit. Sounds like obfuscation for a simple desire to limit some vars life span.

Reply:
I think it's easier to determine the "interface" to the block by its captures and possibly by its return value.

Reply:
It is not an interface, nor does it have a return value. It only limits the scope of some vectors/iters etc. Doing anything beyond scoping it is deliberate obfuscation. But if it really bothers you that much it can just as well not be there, as the main vars are inside loop scopes anyway.